src/components/App/App.tsx

```diff
@@ -52,13 +64,55 @@ export default class App extends React.Component<{}, AppState> {
         console.log('Selected movie id is ' + selectedMovieId);
     };
 
+    onFormSubmit = (movie: Movie) => {
+        console.log('Submitted movie id is ' + movie.title);
+    };
+
+    onCloseDialog = () => {
+        this.setState({ showModal: false });
+    };
+
     getSelectedMovieById = (movieId: number): Movie | undefined => {
         return this.state.movies.find(({ id }) => id === movieId);
     };
 
+    onDeleteMovie = () => {
+        console.log('Movie with id' + this.state.selectedMovieId + 'was deleted');
+    };
+
+    renderModal() {
+        if (!this.state.showModal) return null;
+        let dialogContent = null;
+        let dialogTitle = '';
+
+        if (this.state.isAddMovieModal) {
+            dialogContent = <MovieForm onSubmit={this.onFormSubmit} />;
+            dialogTitle = 'Add Movie';
+        } else if (this.state.isEditMovieModal) {
+            dialogContent = <MovieForm initialMovie={movies[0]} onSubmit={this.onFormSubmit} />;
```

Comment:
And for dialogContent we can use like
<MovieForm onSubmit={this.onFormSubmit}  {...(this.state.movieModalType === "Edit" ? {initialMovies: movies[0]} : {})}  />.
Or we can jus map prop config for movie form in component